src/components/mc/cuda/kernel/mc_cuda_reduce_multi.cu

```diff
@@ -0,0 +1,198 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2020-2021.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "../mc_cuda.h"
+#include "utils/ucc_math.h"
+#ifdef __cplusplus
+}
+#endif
+
+#include <assert.h>
+#include <stdio.h>
+
+#define  DO_OP_MAX(_v1, _v2) (_v1 > _v2 ? _v1 : _v2)
```

Comment:
we have these macros defined for the 3rd time now: maybe move them to some common place ?

Reply:
maybe move it to mc_base.h? or ucc_math.h?

Reply:
Mc math looks good to me